specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,143 @@
+# Table Column [Template]
+
+## Overview
+
+*A high-level description of the column.*
+
+### Background
+
+- *Relevant historical or background information*
+- *Link to relevant work items, related existing issues, etc.*
+
+### Features
+
+*A list of the key features unique to this component. If adding support for an existing Nimble component's capabilities (e.g. as `nimble-table-column-mapping` does for Nimble icons and `nimble-spinner`), consider which parts of that component's API should be exposed, and whether on a per-column or per-cell basis. Any parts of the API explicitly not supported should be called out below as non-goals.*
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+
+## Implementation / Design
+
+*Describe interesting implementation/design decisions, with justifications and alternates considered.*
+
+*Include code snippets showing basic component use and any interesting configurations.*
+
+Typical usage:
+
+```html
+<nimble-table>
+    <nimble-table-column-*>
+        Column Label
+    </nimble-table-column-*>
+</nimble-table>
+```
+
+### API
+
+#### Column Element
+
+*Refer to the [table column type philosophy docs](/packages/nimble-components/src/table/specs/table-columns-hld.md#column-type-philosophy) for column creation policies and naming conventions.*
+
+_Element Name_
+
+-   `nimble-table-column-*`
+
+_Props/Attrs_
+
+-   `attr-name`: string (defaults to ...) - Description of the property
+
+_Content_
+
+-   column label (icon and/or text)
+
+_Events_
+
+*List any events along with event detail interface(s).*
+
+#### Cell View Element
+
+_Element Name_
+
+-   `nimble-table-column-*-cell-view`
+
+_Rendering_
+
+*Describe what the cell view will render.*
+
+#### Group Header View Element
+
+*If an existing group header view element will be used instead of creating a new one:*
+
+    A new element will not be created for the group header view. The column will specify the existing `tableColumnTextGroupHeaderViewTag` as the `groupHeaderViewTag` because the header will contain only text.
+
+*Otherwise:*
+
+_Element Name_
+
+-   `nimble-table-column-*-group-header-view`
+
+_Rendering_
+
+*Describe what the group header view will render.*
+
+### Sorting / Grouping
+
+*Will the column be sortable and/or groupable? If so, which `xxx-field-name` attribute will be assigned to  `operandDataRecordFieldName`? If sortable, which `TableColumnSortOperation` will be used? Describe any interesting use cases.*
+
```

Comment:
Is this not an extra blank line? I'm surprised prettier isn't complaining.

Reply:
It is. We're not running prettier on the top-level specs folder. I manually ran the formatter on it, and it removed the extra line, as well as replaced all `*`s with `_`s for italicizing.

Reply:
Can you make a follow-up PR + tech debt to enable prettier there? Guessing it's not intentional

Reply:
Created #2223